### PR TITLE
FORK: No power EPS Fix

### PR DESCRIPTION
**Pull request by magik6k**

### CODE REVIEW

File: build/forks.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
 package build
 
 const ForkCCM = 1750
+const ForkNoPowerEPSUpdates = 20000
```
Comment (magik6k) [inline build/forks.go]: Probably want sooner

File: chain/stmgr/forks.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import (
+	"context"
+	"github.com/ipfs/go-cid"
+
+	"github.com/filecoin-project/lotus/build"
+)
+
+func (sm *StateManager) handleStateForks(ctx context.Context, pstate cid.Cid, height uint64) (cid.Cid, error) {
+	switch height {
+	case build.ForkNoPowerEPSUpdates: // TODO: +1?
```
Comment (whyrusleeping) [inline chain/stmgr/forks.go]: need to handle null blocks

File: chain/actors/actor_miner.go [hunk 3]
```diff
@@ -394,6 +400,113 @@ func (sma StorageMinerActor) ProveCommitSector(act *types.Actor, vmctx types.VMC
...
+	//
+	// Note: Proving period is a function of sector size; small sectors take less
+	// time to prove than large sectors do. Sector size is selected when pledging.
+	pss, lerr := amt.LoadAMT(types.WrapStorage(vmctx.Storage()), self.ProvingSet)
+	if lerr != nil {
+		return nil, aerrors.HandleExternalError(lerr, "could not load proving set node")
+	}
+
+	if pss.Count == 0 && !self.Active {
+		self.ProvingSet = self.Sectors
```
Comment (whyrusleeping) [inline chain/actors/actor_miner.go]: is there ever a case we could be active, but have zero power?

Comment (whyrusleeping) [reply chain/actors/actor_miner.go]: (pss.Count == 0)

Comment (magik6k) [reply chain/actors/actor_miner.go]: Could fault all sectors

**Pull request merged**